applications/ConvectionDiffusionApplication/tests/cpp_tests/test_eulerian_conv_diff.cpp

```diff
@@ -0,0 +1,250 @@
+
+        // Set the nodal values
+        for (auto &i_node : r_test_model_part.Nodes()) {
+            i_node.FastGetSolutionStepValue(DENSITY) = 1.0;
+            i_node.FastGetSolutionStepValue(CONDUCTIVITY) = 1.0;
+            i_node.FastGetSolutionStepValue(SPECIFIC_HEAT) = 1.0;
+            array_1d<double,3> aux_vel = ZeroVector(3);
+            aux_vel[0] = i_node.X();
+            aux_vel[1] = i_node.Y();
+            i_node.FastGetSolutionStepValue(VELOCITY) = aux_vel;
+        }
+
+        // Test element
+        auto p_element = r_test_model_part.pGetElement(1);
+        Vector RHS = ZeroVector(3);
+        Matrix LHS = ZeroMatrix(3,3);
+        p_element->CalculateLocalSystem(LHS, RHS, r_test_model_part.GetProcessInfo());
+
+        std::vector<double> expected_RHS = {0.0, 0.0, 0.0};
+        std::vector<double> expected_LHS = {1.72752,-0.0928335,-0.0928335,
+                                            -0.197093,1.45074,0.475432,
+                                            -0.197093,0.475432,1.45074};
+
+        for (unsigned int i = 0; i < 3; ++i) {
+            KRATOS_CHECK_NEAR(RHS(i), expected_RHS[i], 1.0e-4);
+            for (unsigned int j = 0; j < 3; ++j) {
+                KRATOS_CHECK_NEAR(LHS(i,j), expected_LHS[i*3+j], 1.0e-4);
+            }
+        }
+    }
+
+    KRATOS_TEST_CASE_IN_SUITE(EulerianConvDiff2D4N, KratosConvectionDiffusionFastSuite)
+    {
+        // Create the test element
+        Model model;
+        auto &r_test_model_part = model.CreateModelPart("TestModelPart");
+        SetEulerianConvDiffTestModelPart(r_test_model_part);
+
+        // Element creation
+        r_test_model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+        r_test_model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+        r_test_model_part.CreateNewNode(3, 1.0, 1.0, 0.0);
+        r_test_model_part.CreateNewNode(4, 0.0, 1.0, 0.0);
+        std::vector<ModelPart::IndexType> elem_nodes{1, 2, 3, 4};
+        r_test_model_part.CreateNewElement("EulerianConvDiff2D4N", 1, elem_nodes, r_test_model_part.pGetProperties(0));
+
+        // Set the nodal values
+        for (auto &i_node : r_test_model_part.Nodes()) {
+            i_node.FastGetSolutionStepValue(DENSITY) = 1.0;
+            i_node.FastGetSolutionStepValue(CONDUCTIVITY) = 1.0;
+            i_node.FastGetSolutionStepValue(SPECIFIC_HEAT) = 1.0;
+            array_1d<double,3> aux_vel = ZeroVector(3);
+            aux_vel[0] = i_node.X();
+            aux_vel[1] = i_node.Y();
+            i_node.FastGetSolutionStepValue(VELOCITY) = aux_vel;
+        }
+
+        // Test element
+        auto p_element = r_test_model_part.pGetElement(1);
+        Vector RHS = ZeroVector(4);
+        Matrix LHS = ZeroMatrix(4,4);
+        p_element->CalculateLocalSystem(LHS, RHS, r_test_model_part.GetProcessInfo());
+
+        std::vector<double> expected_RHS = {0.0, 0.0, 0.0, 0.0};
+        std::vector<double> expected_LHS = {1.49976, 0.494317, -0.232728, 0.494317,
```

Comment:
have you seen [this](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/checks.h#L49-L88)? :D